gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -687,6 +694,12 @@ public GsonBuilder registerTypeAdapter(Type type, Object typeAdapter) {
     return this;
   }
 
+  private boolean isTypeObjectOrJsonElement(Type type) {
+    return (!(type instanceof ParameterizedType) &&
```

Comment:
I think this first check is redundant, isn't it? If `type == Object.class` or `type instanceof Class` then `type` can't be an instance of `ParameterizedType`.

Maybe this would be a little clearer?
```java
return type instanceof Class
    && (type == Object.class || JsonElement.class.isAssignableFrom((Class<?>) type));
```
```